plugins/analysis/client/plugin.go

```diff
@@ -60,135 +49,55 @@ func Run(plugin *node.Plugin) {
-	reportChosenNeighbors(eventDispatchers)
-	reportAcceptedNeighbors(eventDispatchers)
-}
-
-func setupHooks(plugin *node.Plugin, conn *network.ManagedConnection, eventDispatchers *EventDispatchers) {
-	// define hooks ////////////////////////////////////////////////////////////////////////////////////////////////////
-
-	onDiscoverPeer := events.NewClosure(func(ev *discover.DiscoveredEvent) {
-		eventDispatchers.AddNode(ev.Peer.ID().Bytes())
-	})
-
-	onDeletePeer := events.NewClosure(func(ev *discover.DeletedEvent) {
-		eventDispatchers.RemoveNode(ev.Peer.ID().Bytes())
-	})
-
-	onAddChosenNeighbor := events.NewClosure(func(ev *selection.PeeringEvent) {
-		if ev.Status {
-			eventDispatchers.ConnectNodes(local.GetInstance().ID().Bytes(), ev.Peer.ID().Bytes())
-		}
-	})
-
-	onAddAcceptedNeighbor := events.NewClosure(func(ev *selection.PeeringEvent) {
-		if ev.Status {
-			eventDispatchers.ConnectNodes(ev.Peer.ID().Bytes(), local.GetInstance().ID().Bytes())
-		}
-	})
-
-	onRemoveNeighbor := events.NewClosure(func(ev *selection.DroppedEvent) {
-		eventDispatchers.DisconnectNodes(ev.DroppedID.Bytes(), local.GetInstance().ID().Bytes())
-	})
-
-	// setup hooks /////////////////////////////////////////////////////////////////////////////////////////////////////
-
-	discover.Events.PeerDiscovered.Attach(onDiscoverPeer)
-	discover.Events.PeerDeleted.Attach(onDeletePeer)
-	selection.Events.IncomingPeering.Attach(onAddAcceptedNeighbor)
-	selection.Events.OutgoingPeering.Attach(onAddChosenNeighbor)
-	selection.Events.Dropped.Attach(onRemoveNeighbor)
-
-	// clean up hooks on close /////////////////////////////////////////////////////////////////////////////////////////
-
-	var onClose *events.Closure
-	onClose = events.NewClosure(func() {
-		discover.Events.PeerDiscovered.Detach(onDiscoverPeer)
-		discover.Events.PeerDeleted.Detach(onDeletePeer)
-		selection.Events.IncomingPeering.Detach(onAddAcceptedNeighbor)
-		selection.Events.OutgoingPeering.Detach(onAddChosenNeighbor)
-		selection.Events.Dropped.Detach(onRemoveNeighbor)
-
-		conn.Events.Close.Detach(onClose)
-	})
-	conn.Events.Close.Attach(onClose)
-}
-
-func reportKnownPeers(dispatchers *EventDispatchers) {
-	if autopeering.Discovery != nil {
-		for _, peer := range autopeering.Discovery.GetVerifiedPeers() {
-			dispatchers.AddNode(peer.ID().Bytes())
-		}
+		nodeId = local.GetInstance().ID().Bytes()
 	}
-}
 
-func reportChosenNeighbors(dispatchers *EventDispatchers) {
-	if autopeering.Selection != nil {
-		for _, chosenNeighbor := range autopeering.Selection.GetOutgoingNeighbors() {
-			//dispatchers.AddNode(chosenNeighbor.ID().Bytes())
-			dispatchers.ConnectNodes(local.GetInstance().ID().Bytes(), chosenNeighbor.ID().Bytes())
-		}
+	// Get outboundIds (chosen neighbors)
+	outgoingNeighbors := autopeering.Selection.GetOutgoingNeighbors()
+	outboundIds := make([][]byte, len(outgoingNeighbors))
+	for i, neighbor := range outgoingNeighbors {
+		outboundIds[i] = neighbor.ID().Bytes()
```

Comment:
Remark: This does not copy the ID, but returns a reference to the underlying array, so one need be careful not to modify it.

Reply:
Okay, added comments in code for this.